### ISSUE TITLE
PotionEffectArgument throws NoSuchMethodError

### ISSUE THREAD

**Issue by jho5245** [JorelAli/CommandAPI#228]
**CommandAPI version:** _6.10_

**Minecraft version:** _1.17.1_

I made a effect command using PotionEffectArgument
/ceffect entity effect time

**Actual result:**

NoSuchMethodError happens.

```sh
[21:29:24 ERROR]: null
org.bukkit.command.CommandException: Unhandled exception executing 'ceffect jho5245 minecraft:saturation 1' in org.bukkit.craftbukkit.v1_17_R1.command.VanillaCommandWrapper(ceffect)
        at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:169) ~[patched_1.17.1.jar:git-Paper-83]
        at org.bukkit.craftbukkit.v1_17_R1.CraftServer.dispatchCommand(CraftServer.java:821) ~[patched_1.17.1.jar:git-Paper-83]
        at net.minecraft.server.network.ServerGamePacketListenerImpl.handleCommand(ServerGamePacketListenerImpl.java:2165) ~[?:?]
        at net.minecraft.server.network.ServerGamePacketListenerImpl.handleChat(ServerGamePacketListenerImpl.java:1976) ~[?:?]
        at net.minecraft.server.network.ServerGamePacketListenerImpl.handleChat(ServerGamePacketListenerImpl.java:1957) ~[?:?]
        at net.minecraft.network.protocol.game.ServerboundChatPacket.handle(ServerboundChatPacket.java:46) ~[?:?]
        at net.minecraft.network.protocol.game.ServerboundChatPacket.handle(ServerboundChatPacket.java:6) ~[?:?]
        at net.minecraft.network.protocol.PacketUtils.lambda$ensureRunningOnSameThread$1(PacketUtils.java:36) ~[?:?]
        at net.minecraft.server.TickTask.run(TickTask.java:18) ~[patched_1.17.1.jar:git-Paper-83]
        at net.minecraft.util.thread.BlockableEventLoop.doRunTask(BlockableEventLoop.java:149) ~[?:?]
        at net.minecraft.util.thread.ReentrantBlockableEventLoop.doRunTask(ReentrantBlockableEventLoop.java:23) ~[?:?]
        at net.minecraft.server.MinecraftServer.doRunTask(MinecraftServer.java:1348) ~[patched_1.17.1.jar:git-Paper-83]
        at net.minecraft.server.MinecraftServer.shouldRun(MinecraftServer.java:190) ~[patched_1.17.1.jar:git-Paper-83]
        at net.minecraft.util.thread.BlockableEventLoop.pollTask(BlockableEventLoop.java:122) ~[?:?]
        at net.minecraft.server.MinecraftServer.pollTaskInternal(MinecraftServer.java:1327) ~[patched_1.17.1.jar:git-Paper-83]
        at net.minecraft.server.MinecraftServer.pollTask(MinecraftServer.java:1320) ~[patched_1.17.1.jar:git-Paper-83]
        at net.minecraft.util.thread.BlockableEventLoop.managedBlock(BlockableEventLoop.java:132) ~[?:?]
        at net.minecraft.server.MinecraftServer.waitUntilNextTick(MinecraftServer.java:1281) ~[patched_1.17.1.jar:git-Paper-83]
        at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1192) ~[patched_1.17.1.jar:git-Paper-83]
        at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:316) ~[patched_1.17.1.jar:git-Paper-83]
        at java.lang.Thread.run(Thread.java:831) [?:?]
Caused by: java.lang.NoSuchMethodError: 'net.minecraft.world.effect.MobEffectList net.minecraft.commands.arguments.ArgumentMobEffect.a(com.mojang.brigadier.context.CommandContext, java.lang.String)'
        at dev.jorel.commandapi.nms.NMS_1_17_R1.getPotionEffect(NMS_1_17_R1.java:703) ~[?:?]
        at dev.jorel.commandapi.arguments.PotionEffectArgument.parseArgument(PotionEffectArgument.java:57) ~[?:?]
        at dev.jorel.commandapi.CommandAPIHandler.parseArgument(CommandAPIHandler.java:345) ~[?:?]
        at dev.jorel.commandapi.CommandAPIHandler.argsToObjectArr(CommandAPIHandler.java:324) ~[?:?]
        at dev.jorel.commandapi.CommandAPIHandler.lambda$generateCommand$0(CommandAPIHandler.java:306) ~[?:?]
        at com.mojang.brigadier.CommandDispatcher.execute(CommandDispatcher.java:263) ~[patched_1.17.1.jar:git-Paper-83]
        at com.mojang.brigadier.CommandDispatcher.execute(CommandDispatcher.java:177) ~[patched_1.17.1.jar:git-Paper-83]
        at net.minecraft.commands.Commands.a(Commands.java:275) ~[?:?]
        at org.bukkit.craftbukkit.v1_17_R1.command.VanillaCommandWrapper.execute(VanillaCommandWrapper.java:45) ~[patched_1.17.1.jar:git-Paper-83]
        at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:159) ~[patched_1.17.1.jar:git-Paper-83]
        ... 20 more
```

- the command should be successfully performed, giving potion effects to specified entities.

_Edit: Formatting_

**Comment by JorelAli** [comment]
@jho5245 I could not reproduce the issue. Here is the code I wrote:

```java
import org.bukkit.plugin.java.JavaPlugin;
import org.bukkit.potion.PotionEffectType;

import dev.jorel.commandapi.CommandAPICommand;
import dev.jorel.commandapi.arguments.PotionEffectArgument;

public class Main extends JavaPlugin {

	@Override
	public void onEnable() {
		new CommandAPICommand("myeffect")
			.withArguments(new PotionEffectArgument("potion"))
			.executes((sender, args) -> {
				PotionEffectType pot = (PotionEffectType) args[0];
				System.out.println(pot);
			})
			.register();
	}

}
```

```xml
<project xmlns="http://maven.apache.org/POM/4.0.0"
	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
	<modelVersion>4.0.0</modelVersion>
	<groupId>commandapi-228</groupId>
	<artifactId>commandapi-228</artifactId>
	<version>0.0.1-SNAPSHOT</version>

	<repositories>
		<repository>
			<id>jitpack.io</id>
			<url>https://jitpack.io</url>
		</repository>
		<!-- This adds the Spigot Maven repository to the build -->
		<repository>
			<id>spigot-repo</id>
			<url>https://hub.spigotmc.org/nexus/content/repositories/snapshots/</url>
		</repository>
	</repositories>

	<dependencies>
		<dependency>
			<groupId>dev.jorel.CommandAPI</groupId>
			<artifactId>commandapi-core</artifactId>
			<version>6.0.0</version>
			<scope>provided</scope>
		</dependency>
		<!--This adds the Spigot API artifact to the build -->
		<dependency>
			<groupId>org.spigotmc</groupId>
			<artifactId>spigot-api</artifactId>
			<version>1.17.1-R0.1-SNAPSHOT</version>
			<scope>provided</scope>
		</dependency>
	</dependencies>

	<build>
		<sourceDirectory>src</sourceDirectory>
		<plugins>
			<plugin>
				<artifactId>maven-compiler-plugin</artifactId>
				<version>3.8.1</version>
				<configuration>
					<release>16</release>
				</configuration>
			</plugin>
		</plugins>
	</build>
</project>
```

Are you sure you're running a normal Spigot 1.17.1 server and not the Mojang mapped version? By inspection, the `spigot-1.17.1.jar` file which I created using the Spigot BuildTools which I am running has the method you are looking for.

**Comment by jpenilla** [comment]
Should be fixed in Paper 1.17.1 builds 87 and newer.

**Comment by jho5245** [comment]
Yes I updated Paper and it works properly! Thanks for fixing this!
